### PR TITLE
Enhanced List Functionality

### PR DESCRIPTION
**Pull request by openbots-ff**
- Moved existing List Commands to a new Command group.
- Added Create/Write List Item Commands.
- Made necessary modifications to Add List Item Command and Selenium Element Action Command.
- Updated Debug Pane code to account for the new variable types.
- TODO: Revisit commented out code in ScriptVariable in Variable Cleanup Task.

### CODE REVIEW

File: taskt.Commands/List Commands/CreateListCommand.cs [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+using taskt.Core.Utilities.CommonUtilities;
+using taskt.Engine;
+using taskt.UI.CustomControls;
+using Exception = System.Exception;
+
+namespace taskt.Commands
+{
+    [Serializable]
+    [Group("List Commands")]
+    [Description("This command adds an item to an existing list variable.")]
```
Comment (ob-ab) [inline taskt.Commands/List Commands/CreateListCommand.cs]: It seems like you have copied the code from AddListItemCommand and forgot to update the Description.
Please update.

File: taskt.Commands/List Commands/WriteListItemCommand.cs [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+using taskt.Core.Utilities.CommonUtilities;
+using taskt.Engine;
+using taskt.UI.CustomControls;
+using Exception = System.Exception;
+
+namespace taskt.Commands
+{
+    [Serializable]
+    [Group("List Commands")]
+    [Description("This command adds an item to an existing list variable.")]
```
Comment (ob-ab) [inline taskt.Commands/List Commands/WriteListItemCommand.cs]: Please update the Description.

File: taskt.Commands/List Commands/WriteListItemCommand.cs [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    public class WriteListItemCommand : ScriptCommand
```
Comment (ob-ab) [inline taskt.Commands/List Commands/WriteListItemCommand.cs]: I suggest to rename it to **UpdateListItemCommand**.

**Review by ob-ab — CHANGES_REQUESTED** [review]
Please make the requested changes.

File: taskt.Commands/List Commands/CreateListCommand.cs (hunk 1 above)
Comment (openbots-ff) [reply taskt.Commands/List Commands/CreateListCommand.cs]: done

File: taskt.Commands/List Commands/WriteListItemCommand.cs (hunk 2 above)
Comment (openbots-ff) [reply taskt.Commands/List Commands/WriteListItemCommand.cs]: done

File: taskt.Commands/List Commands/WriteListItemCommand.cs (hunk 3 above)
Comment (openbots-ff) [reply taskt.Commands/List Commands/WriteListItemCommand.cs]: done

File: taskt.Commands/List Commands/AddListItemCommand.cs [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+    [Serializable]
+    [Group("List Commands")]
+    [Description("This command adds an item to an existing List variable.")]
+    public class AddListItemCommand : ScriptCommand
+    {
+        [XmlAttribute]
+        [PropertyDescription("List")]
+        [InputSpecification("Provide a List variable.")]
+        [SampleUsage("{vList}")]
+        [Remarks("Any type of variable other than list will cause error.")]
```
Comment (CameronHerwig) [inline taskt.Commands/List Commands/AddListItemCommand.cs]: Should change to "List" to match the other corrections.

**Review by CameronHerwig — CHANGES_REQUESTED** [review]

File: taskt.Commands/List Commands/AddListItemCommand.cs (hunk 4 above)
Comment (openbots-ff) [inline taskt.Commands/List Commands/AddListItemCommand.cs]:
```suggestion
        [Remarks("Any type of variable other than List will cause error.")]
```

Comment (openbots-ff) [reply taskt.Commands/List Commands/AddListItemCommand.cs]: done

Comment (openbots-ff) [reply taskt.Commands/List Commands/AddListItemCommand.cs]: done

**Review by CameronHerwig — APPROVED** [review]

**Review by ob-ab — APPROVED** [review]

**Pull request merged**